Bug 1955595: Add DevPreviewLongLifecycle profile

damemi opened the pull request:
Re: https://issues.redhat.com/browse/WRKLDS-286
This adds a new Beta profile `BetaLongLifecycle` which is similar to `LifecycleAndUtilization` but which does not evict running pods that exceed the latter's 24-hour lifetime limit.

damemi reviewed (commented):
/cc @ingvagabund @soltysh 
I think this is all the work that's needed to add the new profile. PTAL and we can discuss any changes or anything I missed

ingvagabund commented on README.md:
```diff
@@ -74,6 +74,9 @@ The following profiles are currently provided:
 * [`LifecycleAndUtilization`](#LifecycleAndUtilization)
+  
+Along with the following profiles, which are in Beta and may change:
+* [`BetaLongLifecycle`](#BetaLongLifecycle)
```
Why Beta, why not Alpha?

soltysh replied:
Since this is externally visible let's align that with our terminology and use `DevPreview` or `TechPreview`, the former is equivalent of alpha, the latter - beta.

soltysh reviewed (commented):
I agree with Jan, let's change the name to something familiar to OCP, otherwise this lgtm.

damemi commented on manifests/4.8/kube-descheduler-operator.crd.yaml:
```diff
@@ -10,8 +12,6 @@ spec:
     plural: kubedeschedulers
     singular: kubedescheduler
   scope: Namespaced
-  subresources:
-    status: {}
```
This top-level `subresources` field was removed, and replaced with one in each version below when I ran the generator... @ingvagabund will this work or is it the same problem you fixed in https://github.com/openshift/cluster-kube-descheduler-operator/pull/185?

ingvagabund replied:
Yeah, it should work.

soltysh replied:
Yeah, v1beta1 had it here, v1 has it per specific version.

damemi commented:
Updated with the feedback. I also added a check for profile conflicts (to ensure that the new profile isn't enabled alongside the old one, since we have no defined way to resolve that)

ingvagabund commented on pkg/operator/target_config_reconciler.go:
```diff
@@ -242,6 +248,21 @@ func (c *TargetConfigReconciler) manageConfigMap(descheduler *deschedulerv1.Kube
 
+// checkProfileConflicts ensures that multiple profiles aren't redeclared
+// it also checks for various inter-profile conflicts (profiles which should not be enabled simultaneously)
+func checkProfileConflicts(profiles sets.String, profileName deschedulerv1.DeschedulerProfile) error {
```
You wanna write a unit test for it? :P

soltysh replied:
Would be nice, but can be done in a followup.

soltysh reviewed (approved):
/lgtm
